nntrainer/src/databuffer.cpp

```diff
@@ -116,9 +119,16 @@ void DataBuffer::UpdateData(buffer_type type, std::ifstream &file) {
 
           unsigned int id = rangeRandom(0, mark.size() - 1);
           I = mark[id];
+          if (I > max_test || I < 0)
+            std::runtime_error("Error: Test case id cannot exceed maximum number of test");
+
           mark.erase(mark.begin() + id);
 
           int64_t position = (I * input_size + I * class_num) * sizeof(float);
+
+          if (position > file_length)
```

Comment:
position is int64_t but file_length is int. Consider ssize_t.

Reply:
yes. but I think it should be int64_t. I updated file_lenght type.